components/apimgt/org.wso2.carbon.apimgt.rest.api.authenticator/src/main/java/org/wso2/carbon/apimgt/rest/api/authenticator/factories/AuthenticatorAPIFactory.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.apimgt.rest.api.authenticator.factories;
+
+import org.wso2.carbon.apimgt.core.api.KeyManager;
+import org.wso2.carbon.apimgt.core.configuration.APIMConfigurationService;
+import org.wso2.carbon.apimgt.core.dao.SystemApplicationDao;
+import org.wso2.carbon.apimgt.core.dao.impl.DAOFactory;
+import org.wso2.carbon.apimgt.core.exception.APIMgtDAOException;
+import org.wso2.carbon.apimgt.core.exception.KeyManagementException;
+import org.wso2.carbon.apimgt.core.impl.APIManagerFactory;
+import org.wso2.carbon.apimgt.rest.api.authenticator.AuthenticatorService;
+
+public class AuthenticatorAPIFactory {
+    private static AuthenticatorAPIFactory instance = new AuthenticatorAPIFactory();
+    private AuthenticatorService service;
+
+    /**
+     * Private constructor
+     */
+    private AuthenticatorAPIFactory() {
+
+    }
+
+    /**
+     * Get an instance of AuthenticatorAPIFactory
+     * @return instance of AuthenticatorAPIFactory
+     */
+    public static AuthenticatorAPIFactory getInstance() {
+        return instance;
+    }
+
+    /**
+     * Get an instance of AuthenticatorService
+     * @return AuthenticatorService
```

Comment:
Doc comments are not properly formatted with new line after description. Check `getInstance` method as well